acra-censor/handlers/handlers_util_test.go

```diff
@@ -0,0 +1,344 @@
+/*
+Copyright 2018, Cossack Labs Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"fmt"
+	"github.com/xwb1989/sqlparser"
+	"testing"
+)
+
+func TestHandleRangeCondition(t *testing.T) {
+	patterns := []string{
+		// left side value placeholder
+		fmt.Sprintf("select 1 from t where param between %s and 2", ValueConfigPlaceholder),
+		// right side value placeholder
+		fmt.Sprintf("select 1 from t where param between 1 and %s", ValueConfigPlaceholder),
+		// two sides placeholder
+		fmt.Sprintf("select 1 from t where param between %s and %s", ValueConfigPlaceholder, ValueConfigPlaceholder),
+		// two explicit int values
+		"select 1 from t where param between 1 and 2",
+		// two explicit str values
+		"select 1 from t where param between 'qwe' and 'asd'",
+	}
+	parsedPatterns, err := ParsePatterns(patterns)
+	if err != nil {
+		t.Fatal(err)
+	}
+	notMatchableQueries := [][]string{
+		// left side value placeholder
+		[]string{
+			"select 1 from t where param between 'value placeholder' and 3",
+			"select 1 from t where param between 'value placeholder' and 'qwe'",
+			"select 1 from t where param between 'value placeholder' and TRUE",
```

Comment:
this test show that we match these values too (without subquery). we test null/boolean/int/str values in tests to be sure that %%value%% handle it

Reply:
Yes. But it seems that it shouldn't match, because NULL is not exact %%VALUE%% - it's NULL and pattern should be also NULL to match. The same for boolean, subquery, function

Reply:
%%value%% should match any values as literals (int/str/boolean/nullable/bits/etc). and it match now as we can see it from passed tests